flink-yarn/src/main/java/org/apache/flink/yarn/YarnResourceManager.java

```diff
@@ -131,6 +137,11 @@
 
 	private final Resource resource;
 
+	/**
+	 * Executor for starting new yarn containers.
+	 */
+	private final ThreadPoolExecutor startContainerExecutor;
```

Comment:
Wouldn't an `ExecutorService` be enough here?

Reply:
Let's rename the `Executor` into `blockingCallsExecutor`, `yarnCallsExecutor` or `ioExecutor`